plugin/keyring_vault/vault_key.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MYSQL_VAULT_KEY_H
+#define MYSQL_VAULT_KEY_H
+
+#include "keyring_key.h"
+#include "i_serialized_object.h"
+
+namespace keyring {
+
+struct Vault_key : public Key, public ISerialized_object
+{
+  Vault_key(const char *a_key_id, const char *a_key_type, const char *a_user_id,
+      const void *a_key, size_t a_key_len)
+    : Key(a_key_id, a_key_type, a_user_id, a_key, a_key_len)
+    , was_key_retrieved(FALSE)	
+  {}
+
+  Vault_key(const Vault_key &vault_key)
+    : Key(vault_key.key_id.c_str(), vault_key.key_type.c_str(),
+          vault_key.user_id.c_str(), vault_key.key.get(), vault_key.key_len)
+  {
+    this->key_operation = vault_key.key_operation;
+    this->was_key_retrieved = FALSE;
+  }
+  Vault_key()
```

Comment:
The auto-generated default constructor will not be generated as there are explicitly defined other constructors, thus I need to add default constructor "by hand'